posts/2023-02-27-introducing-new-conda.md

```diff
@@ -0,0 +1,53 @@
+
+## Summary
+
+* Same packages as current `analysis3` environments
+* Addresses issues with the installation and maintenance of current `analysis3` environment.
+* Will be maintained in parallel with standard hh5 conda for at least 3 months.
+* Will retain `analysis3-stable/unstable` structure and (roughly) quarterly update time
+* New module name - `conda_concept` - to use on Gadi e.g.
+```
+$ module use /g/data/hh5/public/modules
+$ module load conda_concept/analysis3-unstable
+$ python3
+```
+* `analysis3/22.07`, `analysis3/22.10` and `analysis3/23.01` environments are available - they are exact duplicates of the corresponding existing `analysis3` environments.
+* New module for ARE - `conda_concept/are`.
+![conda-are-settings](../images/conda-are-settings.png)
+* Some catches:
+    * Can't use direct path to interpreter, must use path to the 'scripts' directory instead if a full path is needed (e.g. in `#!/g/data/hh5/.../python3` shbang lines in scripts)
+    * `conda activate` does not work, must load the module.
+* Any problems or questions, contact cws_help@nci.org.au, or leave an issue on the [github page](https://github.com/coecms/cms-conda-singularity/issues).
+
+## Usage
+Usage of the new `conda_concept/analysis3` environments is very similar to the current `conda/analysis3` environments. On the Gadi command line, or in PBS scripts, run
+```
+$ module use /g/data/hh5/public/modules
+$ module load conda_concept/analysis3
+```
+The version naming scheme is carried over from the current `conda/analysis3` environments. The `analysis3-unstable` environment will always be an alias for the most recent environment, and will continue to be updated at roughly 3 monthly intervals. `analysis3` will always be an alias for the next-most recent environment, referred to as the "stable" environment. `module load conda_concept` will always be configured to load the "stable" environment. Specific versions can also be loaded with e.g. `module load conda_concept/analysis3-23.01`. There is also a new module, `conda_concept/are`, designed specifically for use with `ipython` and `jupyter` notebooks. This module loads the "stable" `analysis3` environment, but manipulates the environment such that all of the `analysis3` environments can be used as the kernel for a notebook without having to restart the `jupyter` session.

```

Comment:
I found this paragraph a bit heavy, feel free to dismiss my suggestions on how to lighten it up, but I think it needs some changes
```suggestion
The version naming scheme and is not changing. The `analysis3-unstable` environment will always be an alias for the most recent environment and will continue to be updated at roughly 3 monthly intervals. `analysis3` will remain the most recent "stable" environment.

````{note}
```code
 module load conda_concept
```
 will load the "stable" environment. Specific versions can also be loaded with e.g.
```code
module load conda_concept/analysis3-23.01
```
There is also a new module, `conda_concept/are`, designed specifically for use with `ipython` and `jupyter` notebooks. This module loads the "stable" `analysis3` environment, but manipulates the environment such that all of the `analysis3` environments can be used as the kernel for a notebook without having to restart the `jupyter` session.
````


```

Reply:
I also have a question about the jupyter statement "all of the analysis3 environments can be used as the kernel for a notebook without having to restart the jupyter session" that is my current experience with ARE, regardless of which environment I load

Reply:
Due to each conda env being in its own squashfs, that doesn't work in this environment unless the `conda_concept/are` module is loaded. I will make that more clear

Reply:
Actually, you're right. This should just work. I've modified the 'jupyter' command so that it sneaks in all of the available squashfs file systems regardless of which module is loaded.